### PR TITLE
database/sinkdb/sinkdbtest: new package

### PR DESCRIPTION
**Pull request by jbowens**
Add package with test utilities for creating test sinkdb instances.
This will also prevent tests in separate packages from interacting.

### CODE REVIEW

**Comment by kr** [comment]
It could do what the pgtest package does to GC old databases: name them with the datetime of creation and delete old ones on startup. Note that the finalizer in that package is only meant to close connections to avoid hitting the postgres connection limit. Since sinkdb has no connection limit we prob don't need that here.

**Comment by jbowens** [comment]
Updated to gc old directories using the directory modified time. That should work for our purposes, right?

**Comment by kr** [comment]
> directory modified time

Oh good idea. Yeah that seems reasonable. :eyes:

File: database/sinkdb/sinkdbtest/sinkdbtest.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+func gcDataDirectories() {
+	tempDir := os.TempDir()
+	cutoff := time.Now().Add(-time.Hour * 24)
+	dirents, _ := ioutil.ReadDir(tempDir)
+	for _, dirent := range dirents {
+		if !strings.HasPrefix(dirent.Name(), dataDirectoryPrefix) {
+			continue
+		}
+		if dirent.ModTime().After(cutoff) {
```
Comment (kr) [inline database/sinkdb/sinkdbtest/sinkdbtest.go]: Before?

Comment (kr) [inline database/sinkdb/sinkdbtest/sinkdbtest.go]: Oops my bad.

**Pull request merged**